### ISSUE TITLE
Resumable upload fails with an `The request included a body. Requests of this type must not include a non-empty body` error

### ISSUE THREAD

**Issue by svozza** [aws-amplify/amplify-js#11308]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-js/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-js/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### JavaScript Framework

React

### Amplify APIs

Authentication, Storage

### Amplify Categories

storage

### Environment information

<details>

```
System:
    OS: macOS 12.6.5
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 23.23 MB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.18.0 - ~/.volta/tools/image/node/16.18.0/bin/node
    Yarn: 1.22.19 - ~/.volta/tools/image/yarn/1.22.19/bin/yarn
    npm: 9.6.1 - ~/.volta/tools/image/npm/9.6.1/bin/npm
  Browsers:
    Chrome: 112.0.5615.137
    Firefox: 111.0.1
    Safari: 16.4.1
  npmGlobalPackages:
    corepack: 0.14.1
    npm: 8.19.2


```

</details>

### Describe the bug

Edit: I now think the `resumable` flag is a red herring. The operation fails if my file is larger than 5Mb, regardless of whether this flag exists or not.

When trying to use the resumable upload feature of the Amplify storage library, I receive an error `The request included a body. Requests of this type must not include a non-empty body` error. This appears to be caused by passing the wrong `Input` type to the `CreateMultipartUploadCommand` constructor [here](https://github.com/aws-amplify/amplify-js/blob/0197e92888f9aeb142aefb23c31c8831a296ce51/packages/storage/src/providers/AWSS3UploadTask.ts#L434). If you look at where this params object is created you will see that the `PutObjectCommandInput` constructor is being [used](https://github.com/aws-amplify/amplify-js/blob/0197e92888f9aeb142aefb23c31c8831a296ce51/packages/storage/src/providers/AWSS3Provider.ts#L535). `PutObjectCommandInput` has a field called [Body](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/interfaces/putobjectcommandinput.html#body-3) whereas `CreateMultipartUploadCommand` [doesn't](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-s3/interfaces/createmultipartuploadcommandinput.html), which matches with what the error is complaining about. As a result the operation fails and the object is not uploaded.

### Expected behavior

The object should be uploaded to S3.

### Reproduction steps

Use the code snippet below in an Amplify application to try to upload any file to S3.

### Code Snippet

```javascript
export const uploadObject = async (key, content, level, type) => {
  const blob = new Blob([content], { type });
  const {username} = await Auth.currentAuthenticatedUser();
    return new Promise((res, rej) => {
        Storage.put(key, blob, {
            level: level,
            contentType: type,
            resumable: true,
            metadata: {
                username
            },
            completeCallback: (event) => {
                res(event);
            },
            errorCallback: (err) => {
                console.error(err);
                rej(new Error('We could not complete that action. Please try again'));
            }
        });
  });
};

```

### Log output

<details>

```js
// fff


```

</details>

### Additional information and screenshots

~If I remove the `resumable` boolean, it works as expected.~ If the file is under 5Mb it works as expected. I am using Amplify version `4.3.46`. Something to note is that I am not using Amplify CLI, just the storage libray to interact with an S3 bucket created as part of a CloudFormation script. The CFN to create the bucket is as follows:

```yaml
  AmplifyStorageBucket:
    Type: "AWS::S3::Bucket"
    Properties:
      BucketEncryption:
        ServerSideEncryptionConfiguration:
          - ServerSideEncryptionByDefault:
              SSEAlgorithm: AES256
      CorsConfiguration:
        CorsRules:
          - AllowedHeaders:
              - "*"
            AllowedMethods:
              - GET
              - POST
              - PUT
              - DELETE
            AllowedOrigins:
              - "*"
            ExposedHeaders:
              - x-amz-server-side-encryption
              - x-amz-request-id
              - x-amz-id-2
              - ETag
            Id: WdAmplifyCorsRule
            MaxAge: 3600
      LoggingConfiguration:
        DestinationBucketName: !Ref AccessLogsBucket
        LogFilePrefix: amplify-storage-bucket/
      VersioningConfiguration:
        Status: Enabled

 # ...

  AmplifyStorageBucketPolicy:
    Type: "AWS::S3::BucketPolicy"
    Properties:
      Bucket: !Ref AmplifyStorageBucket
      PolicyDocument:
        Statement:
          - Sid: HttpsOnly
            Effect: Deny
            Principal: "*"
            Action: "*"
            Resource:
              - "Fn::Sub": "arn:aws:s3:::${AmplifyStorageBucket}/*"
              - "Fn::Sub": "arn:aws:s3:::${AmplifyStorageBucket}"
            Condition:
              Bool:
                "aws:SecureTransport": "false"
```

**Comment by nadetastic** [comment]
Hi @svozza thank you for opening this issue.

I've taken a look and haven't been able to reproduce this issue - could you confirm a few things for me?

- Is it happening on all files? Or does it seem to happen only on some files (ex larger files)?
- Also i see that you are using an older version of `aws-amplify` - could you try upgrading to the latest version and see if this still happens?

**Comment by nadetastic** [comment]
@svozza I haven't been able to reproduce this issue, however I'm curious what sort of file types you are uploading. My testing so far has been on text and image files. Could you clarify this?

**Comment by svozza** [comment]
I'm just uploading a JSON file with a content type `application/json'.

**Comment by svozza** [comment]
Btw, here is the code that I used to workaround the issue. There might be some subtle differences between how the `lib-storage` and Amplify write to S3:

```js
export const uploadObject = async (key, content, level, type) => {
    const Body = new Blob([content], { type });
    
    const [{username}, {accessKeyId, secretAccessKey, sessionToken, identityId}] = await Promise.all([
        Auth.currentAuthenticatedUser(),
        Auth.currentUserCredentials()
    ]);

    const {bucket: Bucket, region} = Storage._config.AWSS3;
    
    // there is a specific folder structure amplify  storage expects
    const Key = level === 'private'
        ? `private/${identityId}/${key}`
        : `public/${key}`;

    const parallelUploads3 = new Upload({
        client: new S3Client({
            credentials: {
                accessKeyId,
                secretAccessKey,
                sessionToken,
            },
            region
        }),
        params: {
            ContentType: type, Bucket, Key, Body, Metadata: {username}
        },
    });

    return parallelUploads3.done()
        .catch((err) => {
            console.error(err);
            throw new Error('We could not complete that action. Please try again');
        });
};```

**Comment by nadetastic** [comment]
@svozza thanks for the additional context, i was able to reproduce this and it appears to happen if you specify `contentType: 'application/json'`. Additionally it seems to happen on any file size if `resumable:true` is also enabled.
I'm discussing this with the team and will follow up with you.

For now, the work around you have can work, but you can also omit `contentType` when using Amplify Storage as it should still work fine - unless you have a strong requirement to specify it.

**Comment by svozza** [comment]
Thank you. Removing the content type works for me, although if I don't include the the `resumable` boolean it fails even if I don't supply the content type..

**Comment by nadetastic** [comment]
Hi @simara001, what is the file type you are uploading?

**Comment by simara001** [comment]
> Hi @simara001, what is the file type you are uploading?

@nadetastic  I've tried with zip/dmg/mp4 videos. Same result.

**Comment by LizDodion** [comment]
Also having the same issue with any file above 5mb. I am using amplify 5.1.2

**Comment by nadetastic** [comment]
@LizDodion curious if you have experienced this with the latest version of `aws-amplify`?

**Comment by nadetastic** [comment]
Hi @LizDodion @simara001 @svozza this issue should now be fixed with the latest version of amplify `5.3.4`.

Let me know if you are still seeing this when using this latest version.
